src/AzureReadOnlyStore.cs

```diff
@@ -61,30 +63,30 @@ public bool IsSameStore(IReadOnlyStore other)
             var blobPrefix = $"{Realm}/{prefix:X2}";
             var count = 0;
 
-            var token = default(BlobContinuationToken);
+            var token = default(string);
             do
             {
-                var result = await Persistent.ListBlobsSegmentedAsync(
-                    prefix: blobPrefix,
-                    useFlatBlobListing: true,
-                    blobListingDetails: BlobListingDetails.Metadata,
-                    maxResults: null,
-                    currentToken: token,
-                    options: null,
-                    operationContext: null,
-                    cancellationToken: cancel).ConfigureAwait(false);
-
-                foreach (var item in result.Results)
+                var result = await Persistent.GetBlobsAsync(BlobTraits.Metadata,
+                BlobStates.None,
+                blobPrefix,
+                cancel).AsPages(token)
+                .ToListAsync(cancel)
```

Comment:
@EslaMx7 I would suggest to remove `.ToListAsync` and use `async foreach`  https://docs.microsoft.com/en-us/archive/msdn-magazine/2019/november/csharp-iterating-with-async-enumerables-in-csharp-8  and https://docs.microsoft.com/en-us/dotnet/api/azure.asyncpageable-1?view=azure-dotnet

@VictorNicollet Do you confirm?

Reply:
@vermorel, @EslaMx7 using `await foreach` would indeed be the best solution here. It will require upgrading Lokad.ContentAddr.Azure.csproj to .NET Standard 2.1 (it's currently on 2.0), which I think is possible without breaking anything, so let's do it.

Reply:
@vermorel @VictorNicollet Sure this would be much better, just wanted to keep minimum changes. I will do the required updates.